tests/model_registry/rbac/conftest.py

```diff
@@ -0,0 +1,172 @@
+    Function scope because token is temporary and tied to the SA for that test.
+    """
+    sa_name = service_account.name
+    namespace = service_account.namespace
+    LOGGER.info(f"Retrieving token for ServiceAccount: {sa_name} in namespace {namespace}")
+    try:
+        cmd = f"oc create token {sa_name} -n {namespace} --duration={DEFAULT_TOKEN_DURATION}"
+        LOGGER.debug(f"Executing command: {cmd}")
+        res, out, err = run_command(command=shlex.split(cmd), verify_stderr=False, check=True, timeout=30)
+        token = out.strip()
+        if not token:
+            raise ValueError("Retrieved token is empty after successful command execution.")
+
+        LOGGER.info(f"Successfully retrieved token for SA '{sa_name}'")
+        return token
+
+    except Exception as e:  # Catch all exceptions from the try block
+        error_type = type(e).__name__
+        log_message = (
+            f"Failed during token retrieval for SA '{sa_name}' in namespace '{namespace}'. "
+            f"Error Type: {error_type}, Message: {str(e)}"
+        )
+        if isinstance(e, subprocess.CalledProcessError):
+            # Add specific details for CalledProcessError
+            # run_command already logs the error if log_errors=True and returncode !=0,
+            # but we can add context here.
+            stderr_from_exception = e.stderr.strip() if e.stderr else "N/A"
+            log_message += f". Exit Code: {e.returncode}. Stderr from exception: {stderr_from_exception}"
+        elif isinstance(e, subprocess.TimeoutExpired):
+            timeout_value = getattr(e, "timeout", "N/A")
+            log_message += f". Command timed out after {timeout_value} seconds."
+        elif isinstance(e, FileNotFoundError):
+            # This occurs if 'oc' is not found.
+            # e.filename usually holds the name of the file that was not found.
+            command_not_found = e.filename if hasattr(e, "filename") and e.filename else shlex.split(cmd)[0]
+            log_message += f". Command '{command_not_found}' not found. Is it installed and in PATH?"
+
+        LOGGER.error(log_message, exc_info=True)  # exc_info=True adds stack trace to the log
+        raise
+
+
+# --- RBAC Fixtures ---
+
+
+@pytest.fixture(scope="function")
+def mr_access_role(
+    admin_client: DynamicClient,
+    model_registry_namespace: str,
+    sa_namespace: Namespace,
+) -> Generator[Role, None, None]:
+    """
+    Creates the MR Access Role using direct constructor parameters and a context manager.
+    """
+    role_name = f"registry-user-{MR_INSTANCE_NAME}-{sa_namespace.name[:8]}"
+    LOGGER.info(f"Defining Role: {role_name} in namespace {model_registry_namespace}")
+
+    role_rules: List[Dict[str, Any]] = [
+        {
+            "apiGroups": [""],
+            "resources": ["services"],
+            "resourceNames": [MR_INSTANCE_NAME],  # Grant access only to the specific MR service object
+            "verbs": ["get"],
```

Comment:
should we pass the verbs as input param so to have more flexibility?

Reply:
in our case we only really care about `get` on the MR Instance, so I don't think it's really needed. If we were to generalize this to create any Role then I'd agree with you